tests/integration/golang/mlflow/artifact/get_artifact_s3_test.go

```diff
@@ -0,0 +1,269 @@
+	"github.com/G-Research/fasttrackml/pkg/api/mlflow/api/request"
+	"github.com/G-Research/fasttrackml/pkg/api/mlflow/dao/models"
+	"github.com/G-Research/fasttrackml/tests/integration/golang/fixtures"
+	"github.com/G-Research/fasttrackml/tests/integration/golang/helpers"
+)
+
+type GetArtifactS3TestSuite struct {
+	suite.Suite
+	s3Client           *s3.Client
+	runFixtures        *fixtures.RunFixtures
+	serviceClient      *helpers.HttpClient
+	experimentFixtures *fixtures.ExperimentFixtures
+}
+
+func TestGetArtifactS3TestSuite(t *testing.T) {
+	suite.Run(t, new(GetArtifactS3TestSuite))
+}
+
+func (s *GetArtifactS3TestSuite) SetupTest() {
+	s3Client, err := helpers.NewS3Client(helpers.GetS3EndpointUri())
+	assert.Nil(s.T(), err)
+	s.s3Client = s3Client
+
+	s.serviceClient = helpers.NewMlflowApiClient(helpers.GetServiceUri())
+
+	experimentFixtures, err := fixtures.NewExperimentFixtures(helpers.GetDatabaseUri())
+	assert.Nil(s.T(), err)
+	s.experimentFixtures = experimentFixtures
+
+	runFixtures, err := fixtures.NewRunFixtures(helpers.GetDatabaseUri())
+	assert.Nil(s.T(), err)
+	s.runFixtures = runFixtures
+}
+
+func (s *GetArtifactS3TestSuite) Test_Ok() {
+	defer func() {
+		assert.Nil(s.T(), s.experimentFixtures.UnloadFixtures())
+	}()
+
+	testData := []struct {
+		name   string
+		bucket string
+	}{
+		{
+			name:   "TestWithBucket1",
+			bucket: "bucket1",
+		},
+		{
+			name:   "TestWithBucket2",
+			bucket: "bucket2",
+		},
+	}
+
+	for _, tt := range testData {
+		s.T().Run(tt.name, func(t *testing.T) {
+			// create test experiment
+			experiment, err := s.experimentFixtures.CreateExperiment(context.Background(), &models.Experiment{
+				Name:             fmt.Sprintf("Test Experiment In Bucket %s", tt.bucket),
+				LifecycleStage:   models.LifecycleStageActive,
+				ArtifactLocation: fmt.Sprintf("s3://%s/1", tt.bucket),
+			})
+			assert.Nil(s.T(), err)
+
+			// create test run
+			runID := strings.ReplaceAll(uuid.New().String(), "-", "")
+			run, err := s.runFixtures.CreateRun(context.Background(), &models.Run{
+				ID:             runID,
+				Status:         models.StatusRunning,
+				SourceType:     "JOB",
+				ExperimentID:   *experiment.ID,
+				ArtifactURI:    fmt.Sprintf("%s/%s/artifacts", experiment.ArtifactLocation, runID),
+				LifecycleStage: models.LifecycleStageActive,
+			})
+			assert.Nil(s.T(), err)
+
+			// upload artifact root object to S3
+			putObjReq := &s3.PutObjectInput{
+				Key:    aws.String(fmt.Sprintf("1/%s/artifacts/artifact.file", runID)),
+				Body:   strings.NewReader(`content`),
+				Bucket: aws.String(tt.bucket),
+			}
+			_, err = s.s3Client.PutObject(context.Background(), putObjReq)
+			assert.Nil(s.T(), err)
+
+			// upload artifact subdir object to S3
+			putObjReq = &s3.PutObjectInput{
+				Key: aws.String(fmt.Sprintf(
+					"1/%s/artifacts/artifact.subdir/artifact.file",
+					runID),
+				),
+				Body:   strings.NewReader(`subdir-object-content`),
```

Comment:
for future PR: let's get rid of those backtick strings here and in the ListArtifacts tests

Reply:
done in #405